nrf_rpc/nrf_rpc.c

```diff
@@ -597,19 +598,44 @@ static int init_packet_handle(struct header *hdr, const struct nrf_rpc_group **g
 	}
 
 	/* Check for the corresponding local group and initialize it. */
-	*group = remote_group_init(hdr->src_group_id, init_data.strid, init_data.strid_len);
+	*group = group_from_strid(init_data.strid, init_data.strid_len);
 	if (*group == NULL) {
+		NRF_RPC_ERR("Remote group does not match local group");
 		NRF_RPC_ASSERT(false);
 		return -NRF_EFAULT;
 	}
 
-	initialized_group_count++;
-	if (initialized_group_count == group_count) {
+	group_data = (*group)->data;
+	first_init = group_data->dst_group_id == NRF_RPC_ID_UNKNOWN;
+	group_data->dst_group_id = hdr->src_group_id;
+	bool wait_on_init = (*group)->wait_on_init;
+	nrf_rpc_group_bound_handler_t bound_handler = (*group)->bound_handler;
+
+	NRF_RPC_DBG("Found corresponding local group. Remote id: %d, Local id: %d",
+		    hdr->src_group_id, group_data->src_group_id);
+
+	if (bound_handler != NULL) {
```

Comment:
Would it make sense to only call it once on `first_init`? Or the idea is to also use it for notifying about reboots?

Reply:
My idea was to call it each time the remote reboots and binds. The local application can use the callback to reset its own state.